internal/cluster/refresh_status_preprocessor.go

```diff
@@ -124,6 +137,72 @@ func (r *refreshPreprocessor) preprocess(ctx context.Context, c *clusterPreproce
 	return stateMachineInput, validationsOutput, nil
 }
 
+// recalculateOperatorDependencies calculates the operator dependencies and updates the database and the passed cluster
+// accordingly.
+func (r *refreshPreprocessor) recalculateOperatorDependencies(c *clusterPreprocessContext) error {
+	previousOperators := c.cluster.MonitoredOperators
+	currentOperators, err := r.operatorsAPI.ResolveDependencies(c.cluster, c.cluster.MonitoredOperators)
+	if err != nil {
+		return errors.Wrapf(
+			err,
+			"failed to resolve operator dependencies for cluster '%s'",
+			c.clusterId,
+		)
+	}
+	var addedOperators, updatedOperators, deletedOperators []*models.MonitoredOperator
+	for _, currentOperator := range currentOperators {
+		if currentOperator.ClusterID == "" {
+			currentOperator.ClusterID = c.clusterId
+		}
+		previousOperator := operatorcommon.GetOperator(previousOperators, currentOperator.Name)
+		if previousOperator != nil {
+			if !reflect.DeepEqual(currentOperator, previousOperator) {
+				updatedOperators = append(updatedOperators, currentOperator)
+			}
+		} else {
+			addedOperators = append(addedOperators, currentOperator)
+		}
+	}
+	for _, previousOperator := range previousOperators {
+		if !operatorcommon.HasOperator(currentOperators, previousOperator.Name) {
+			deletedOperators = append(deletedOperators, previousOperator)
+		}
+	}
+	for _, addedOperator := range addedOperators {
+		err = c.db.Save(addedOperator).Error
```

Comment:
@rccrdpccl Saving the cluster changes directly to DB will cause inconsistent values in the elastic, right?